### ISSUE TITLE
Influxdb2  Unauthorized

### ISSUE THREAD

**Issue by a1ekseev** [vitaliy-sk/keenetic-grafana-monitoring#5]
Hi!
I have a problem when connecting to Influxdb2.

```Python 
2021-10-31 09:59:08,607 - influxdb_writter.py - INFO - Connecting to InfluxDB: 172.19.0.4
Traceback (most recent call last):
  File "/home/keentic_influxdb_exporter.py", line 108, in <module>
    infuxdb_writer = InfuxWriter(config['influxdb'])
  File "/home/influxdb_writter.py", line 15, in __init__
    self.init_database()
  File "/home/influxdb_writter.py", line 22, in init_database
    if db_name not in self._client.get_list_database():
  File "/root/.local/lib/python3.8/site-packages/influxdb/client.py", line 704, in get_list_database
    return list(self.query("SHOW DATABASES").get_points())
  File "/root/.local/lib/python3.8/site-packages/influxdb/client.py", line 521, in query
    response = self.request(
  File "/root/.local/lib/python3.8/site-packages/influxdb/client.py", line 378, in request
    raise InfluxDBClientError(err_msg, response.status_code)
influxdb.exceptions.InfluxDBClientError: 401: {"code":"unauthorized","message":"Unauthorized"}
```
I noticed that this library is used - [InfluxDB-Python](https://influxdb-python.readthedocs.io/en/latest/include-readme.html#id2)

Where it clearly states:
> Note: This library is for use with InfluxDB 1.x. For connecting to InfluxDB 2.x instances, please use the the [influxdb-client-python](https://github.com/influxdata/influxdb-client-python) client.

I saw that Influxdb2 integration was added in [version 1.1.0](https://github.com/vitaliy-sk/keenetic-grafana-monitoring/releases/tag/1.1.0)

My build:
```docker-compose
  influxdb2:
    image: influxdb:2.0-alpine
    container_name: influxdb2
    restart: unless-stopped
    volumes:
      - ./influxdb2/data:/var/lib/influxdb2
    env_file:
      - influxdb2.env

  keenetic-monitor:
    build: ./keenetic-monitor/build
    image: keenetic-monitor:latest
    container_name: keenetic-monitoring
    restart: unless-stopped
    environment:
      - TZ=Europe/Moscow
    volumes:
      - ./keenetic-monitor/config/config.ini:/home/config/config.ini:ro
      - ./keenetic-monitor/config/metrics.json:/home/config/metrics.json:ro
```

**Comment by vitaliy-sk** [comment]
@a1ekseev yes, this is a mistake in release notes, currently it support influx 1.8.x
with influx v2 you can try use legacy auth https://docs.influxdata.com/influxdb/v2.0/reference/cli/influx/v1/auth/
but I can recommend to use support influx 1.8.x for now

**Comment by a1ekseev** [comment]
> @a1ekseev yes, this is a mistake in release notes, currently it support influx 1.8.x with influx v2 you can try use legacy auth https://docs.influxdata.com/influxdb/v2.0/reference/cli/influx/v1/auth/ but I can recommend to use support influx 1.8.x for now

I use influxdb2 for homeassistant, I do not want to deploy a separate version of influxdb.

Is it expensive to add full support for influxdb2?

**Comment by a1ekseev** [comment]
> @a1ekseev yes, this is a mistake in release notes, currently it support influx 1.8.x with influx v2 you can try use legacy auth https://docs.influxdata.com/influxdb/v2.0/reference/cli/influx/v1/auth/ but I can recommend to use support influx 1.8.x for now

New problem:
```Python
influxdb.exceptions.InfluxDBClientError: not implemented: CREATE DATABASE
```

**Comment by vitaliy-sk** [comment]
@a1ekseev which influxdb version are you using at the end?

**Comment by a1ekseev** [comment]
> @a1ekseev which influxdb version are you using at the end?

influxdb2

**Comment by vitaliy-sk** [comment]
@a1ekseev I added influxdb 2 support in separate branch
pls try _techh/keenetic-grafana-monitoring:candidate_  docker image
config sample for influxdb 2 is [here](https://github.com/vitaliy-sk/keenetic-grafana-monitoring/blob/candidate/config/config.sample-influx-2.x.ini)
please let me know if it's working fine

**Comment by a1ekseev** [comment]
> @a1ekseev I added influxdb 2 support in separate branch pls try _techh/keenetic-grafana-monitoring:candidate_ docker image config sample for influxdb 2 is [here](https://github.com/vitaliy-sk/keenetic-grafana-monitoring/blob/candidate/config/config.sample-influx-2.x.ini) please let me know if it's working fine

Great, thanks, I'll try to set it up soon!

**Comment by a1ekseev** [comment]
> @a1ekseev I added influxdb 2 support in separate branch pls try _techh/keenetic-grafana-monitoring:candidate_ docker image config sample for influxdb 2 is [here](https://github.com/vitaliy-sk/keenetic-grafana-monitoring/blob/candidate/config/config.sample-influx-2.x.ini) please let me know if it's working fine

```log
2021-11-13 19:13:53,490 - influxdb_writter.py - INFO - Connecting to InfluxDB: http://influxdb2:8086
2021-11-13 19:13:53,490 - keentic_influxdb_exporter.py - INFO - Connecting to router: http://10.10.10.1:80
2021-11-13 19:13:53,490 - keentic_influxdb_exporter.py - INFO - Configuring metric: processes
2021-11-13 19:13:53,564 - keentic_influxdb_exporter.py - INFO - Configuring metric: ip hotspot
2021-11-13 19:13:53,749 - keentic_influxdb_exporter.py - INFO - Configuring metric: interface
2021-11-13 19:13:53,904 - keentic_influxdb_exporter.py - INFO - Configuring metric: system
2021-11-13 19:13:54,031 - keentic_influxdb_exporter.py - INFO - Configuring metric: internet status
2021-11-13 19:13:54,107 - keentic_influxdb_exporter.py - INFO - Configuring metric: media
2021-11-13 19:13:54,184 - keentic_influxdb_exporter.py - INFO - Configuring metric: interface stat
2021-11-13 19:13:54,251 - keentic_influxdb_exporter.py - INFO - Configuring metric: interface stat
2021-11-13 19:13:54,319 - keentic_influxdb_exporter.py - INFO - Configuring metric: interface stat
2021-11-13 19:13:54,383 - keentic_influxdb_exporter.py - INFO - Configuring metric: interface stat
2021-11-13 19:13:54,446 - keentic_influxdb_exporter.py - INFO - Configuring metric: interface stat
2021-11-13 19:13:54,510 - keentic_influxdb_exporter.py - INFO - Configuration done. Start collecting with interval: 30 sec
```

It's works!
It remains to rewrite the queries on Flux.
Flux by the way beta in prometheus.

**Comment by vitaliy-sk** [comment]
@a1ekseev Good, please share your dashboard with flux queries

**Comment by a1ekseev** [comment]
> @a1ekseev Good, please share your dashboard with flux queries

I will! As soon as I understand how to work queries through Flux.

**Comment by vitaliy-sk** [comment]
Fixed in [2.0.0](https://github.com/vitaliy-sk/keenetic-grafana-monitoring/releases/tag/2.0.0)
